xbmc/platform/win32/WIN32Util.cpp

```diff
@@ -1424,7 +1424,7 @@ HDR_STATUS CWIN32Util::GetWindowsHDRStatus()
               if (getColorInfo.advancedColorEnabled)
                 advancedColorEnabled = true;
 
-              if (getColorInfo.advancedColorSupported)
+              if (getColorInfo.advancedColorSupported && !getColorInfo.wideColorEnforced)
```

Comment:
This is a ugly hack that mostly will break and cause issues in all others systems. Please not do that.

`getColorInfo.advancedColorSupported` by itself signals that display _supports_ HDR  (enabled or not enabled) you can not add other conditions here.

HDR may be supported but not enabled in Windows as you know (and even not enabled needs to be detected as supported).

`getColorInfo.advancedColorEnabled` by itself signals that HDR is _enabled_ this by logic includes that is supported.

Reply:
> AdvancedColorSupported. This indicates that your installed display driver and display support HDR10.
> 
> AdvancedColorEnabled. This indicates that HDR is currently turned on for your display.

https://support.microsoft.com/en-us/windows/hdr-settings-in-windows-2d767185-38ec-7fdc-6f97-bbc6c5ef24e6

Reply:
Also original issue seems is not working tone mapping (dark picture) but is for unrelated reason. In all systems tone mapping should work and works with HDR_OFF:

// HDR_STATUS::HDR_OFF
advancedColorSupported = true;
advancedColorEnabled = false;

I mean the required condition is HDR_OFF only (not HDR_UNSUPPORTED)

=> Display supports HDR but is OFF then use tone mapping.

Reply:
The OP had two separate issues, screen HDR capabilities and broken dxva processor conversion (the reason for broken tonemapping). This deals only with the first.

I'd rather trust API documentation than description of user settings, which can have some processing on top of the API.

SDR/wide color support is new in Windows 11, see "Windows 11, version 22H2: SDR displays with Advanced Color" , "This topic assumes that your app is targeting Windows 10, version 2004 (or later) for HDR displays, and the Windows 11, version 22H2 release (or later) for SDR displays." "Option 1: AdvancedColorInfo" in https://learn.microsoft.com/en-us/windows/win32/direct3darticles/high-dynamic-range to understand what AdvancedColor means to Microsoft.

The Win32 structure has unclear documentation, only in the SDK header.
See related WinRT API, https://learn.microsoft.com/en-us/uwp/api/windows.graphics.display.advancedcolorkind?view=winrt-22621 and https://learn.microsoft.com/en-us/uwp/api/windows.graphics.display.advancedcolorinfo?view=winrt-22621

AdvancedColor doesn't mean only HDR. Right from the start they planned the WideColorGamut value as a different advanced color value, with SDR transfer and BT2020 colors.

The Win32 structure exposes bits so it was not possible to represent the same way as WinRT 0, 1, 2. Instead:

advancedColorSupported = 0 means SDR
advancedColorSupported = 1 and wideColorEnforced = 1 means SDR with wide colors
advancedColorSupported = 1 and wideColorEnforced = 0 means PQ transfer + wide colors (the usual HDR)

This is supported by the log files in the ticket for my special builds that logged all the bits of DISPLAYCONFIG_GET_ADVANCED_COLOR_INFO.

I agree that for clarity I should have changed the Kodi variable name to "hdrSupported" but the condition is good. I'll update.

I know it's kind of risky and wouldn't mind merging this in early v22 to have plenty of feedback instead.

An alternative is to switch Kodi build in order to allow calling directly the native WinRT API, there are a few others that could be interesting (override blue light filter, dynamically detect dark mode, there will be more in the future)

Reply:
> advancedColorSupported = 0 means SDR
> advancedColorSupported = 1 and wideColorEnforced = 1 means SDR with wide colors
> advancedColorSupported = 1 and wideColorEnforced = 0 means PQ transfer + wide colors (the usual HDR)

In my desktop PC with Nvidia and SDR only monitor is:

![color-info](https://github.com/xbmc/xbmc/assets/58434170/e2e5f66b-2278-40db-b1c8-2b5314d8a3ef)

Then is not clear...

advancedColorSupported = 0 and wideColorEnforced = 1 means ?

Reply:
> I know it's kind of risky and wouldn't mind merging this in early v22 to have plenty of feedback instead.

I agree

Reply:
Without proper documentation it's difficult. It seems that usage of wideColorEnforced is new in Win11.
It's always 0 for me in Win 10 and all vendors.

the rule "advancedColorSupported = 0 means SDR" still works for your config.

I would have liked to know the results of the WinRT API for the OP (and you) but didn't find a quick way to do it.

Reply:
My concern is that a display that supports both HDR and WCG probably _may_ have:

advancedColorSupported = 1
wideColorEnforced = 1

With current PR will broke HDR --> Detected as not HDR capable then Kodi is unable to toggle Windows HDR on

Reply:
For OP both flags =1 mean SDR with WCG (HDR setting of the screen turned off)
His screen reports advancedColorSupported = 1 wideColorEnforced = 0 for "true" HDR, the same as me, and you check your system too.
But it's sample size of 1.